ospfd/ospfd.c

```diff
@@ -798,6 +798,22 @@ static void ospf_finish_final(struct ospf *ospf)
 			ospf_vrf_unlink(ospf, vrf);
 	}
 
+	for (i = 0; i < ZEBRA_ROUTE_MAX + 1; i++) {
+		while (ospf->external[i] && !list_isempty(ospf->external[i])) {
```

Comment:
This is redundant with lines 759 - 780 of the same function which normally clean all external routes. Why looping again external routes ? Is there use cases where the first attempt to clean external routes is not sufficient and some external routes remains active in Zebra after ospfd finish ?

Reply:
The list data structures are not cleaned up.  so this does that, notice that ospf_external_del is not called above.

Reply:
OK. But why not simply add call to ospf_external_del() around line 779 i.e. in the block which remove external routes ?

Reply:
yep fied